<?xml version='1.0'?>
<launch>

 <arg name="base_global_planner" default="navfn/NavfnROS" />
	<!-- Frenet Planner -->

	<node pkg = "frenet_planner" type = "frenet_planner" name = "frenet_planner" output = "screen" >
		<rosparam file = "$(find frenet_planner)/cfg/frenet_planner.yaml" command = "load" />
  </node>
  
  
  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter_right">
        <remap from="scan" to="prius/front_right_laser/scan" />
        <remap from="scan_filtered" to="scan_filtered_right" />
        
        <rosparam file = "$(find frenet_planner)/cfg/laser_filter.yaml" command = "load" />
  </node>

  <node pkg="laser_filters" type="scan_to_scan_filter_chain" output="screen" name="laser_filter_left">
    <remap from="scan" to="/prius/front_left_laser/scan" />
    <remap from="scan_filtered" to="/scan_filtered_left" />
    <rosparam file = "$(find frenet_planner)/cfg/laser_filter.yaml" command = "load" />
</node>
    

	<node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="use_sim_time" value="true" />
    <param name="base_global_planner" value="$(arg base_global_planner)" />
    <!-- <param name="base_local_planner" value="$(arg base_local_planner)" />    -->
    <!-- <rosparam file="$(find car_demo)/cfg/teb_local_planner_params.yaml" command="load"/>    -->
    <rosparam file="$(find frenet_planner)/cfg/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find frenet_planner)/cfg/costmap_common.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find frenet_planner)/cfg/costmap_local.yaml" command="load" />
    <rosparam file="$(find frenet_planner)/cfg/costmap_global.yaml" command="load" />
  </node>

  <!-- Run the tracker node -->
  <node pkg= "tracking_control" name="tracker" type="pure_pursuit.py" output="screen"/>
  
  <!-- Run the controller node --> 
  <node pkg= "tracking_control" name="controller" type="PID_MIT_velocity_controller.py" output="screen"/>

</launch>